### ISSUE TITLE
Why didn't `release-me` pick up #5905 for v6.6.0?

### ISSUE THREAD

**Issue by vito** [concourse/concourse#6158]
This PR wasn't detected: https://github.com/concourse/concourse/pull/5905 - why?

This PR was a substantial change noticeable by end users - is there a bug with the release note generation that caused it not to be included?

Maybe something to do with it being a revert?

**Comment by aoldershaw** [comment]
We investigated and found that it is probably caused by the fact that `release-me` fetches all of the prs that are merged **after the latest release of the repository**. This causes problems when there are patch releases made between minor releases. In our case, since the latest release before v6.6.0 was v6.5.1, only the prs that were merged after v6.5.1 were included in the v6.6.0 release. This means that everything merged between v6.5.0 and v6.5.1 was not included in the v6.6.0 release notes.

```
             5905 merged           PRs used for release notes
<--- 6.5.0 ----------------- 6.5.1 -------------------------- 6.6.0 ---->
```

Possible solution:
- Skip over patch releases: when we are releasing a new minor/major version, we should skip over patch releases for figuring out the starting point of which to start fetching the prs to be in the release note (`FetchCommitsFromReleases`)
- Don't include prs that were released in patches.

**Comment by clarafu** [comment]
Scenario for confirming missing pull requests

1. Merged PR#1 into master
2. Release v1.0.0
	- PR#1
3. Merge PR#2 into master
4. Create release/1.0.0x using PR#2
3. Merge PR#3 into master
4. Release v1.0.1
    - PR#2
5. Merge PR#4 into master
6. Release v1.1.0
    - PR#3
    - PR#4

**Comment by clarafu** [comment]
After some discussion, I decided that it might be best to just have major/minor releases to include pull requests that were mentioned in patches. My decision came from multiple reasons:
1. Implementing the part where we ignore pull requests that are already included in patch releases turned out to be quite a lot of hacky work. The difficult part was because we need some way to fetch all the pull requests that were released by a patch release. I thought of two different ways to implement this in the release-me app. One way was to either have a new flag or to construct a branch name for the **previous release branch** used to create the patch releases. This previous release branch will then be used to fetch the pull requests that were used by the patch releases. The second option was to manually parse the release notes of the patch release and grab out all the pull requests mentioned in it. This is very hacky but is definitely much easier than the first option. While the first option seems like the better candidate, the odd part is needing to come up with the previous release branch name and also all the extra work involved in fetching the pull requests from all the patch releases that were released between the last major/minor release. As a result, none of the options I thought of were really great candidates.
2. At first, I thought that it might be weird having both patch releases and major/minor releases mention the same pull request within their release notes. But after I thought about it some more, it might actually be useful to have them be condensed. This is because if you look at the releases within Concourse, the release notes for patch releases typically only contain one or two pull requests and are quite spread out when you have multiple patches within a major/minor version. When I am looking for a version, I usually skip over the patch releases and just look at the notes for major/minor releases. So if we were to include the pull requests that were a part of patch releases in the release notes for the next major/minor release, that would mean that our users would not need to go back to read the release notes for patch releases in order to see if a bug that was shipped in a patch release would be included in the next major/minor release.

With that in mind, the work that needs to go into this is determining when to start fetching pull requests for the release note generation. If the next release is:
- major/minor: start fetching pull requests from the last major/minor release
- patch: start fetching pull requests from the last release (regardless if it is a patch/major/minor)
For this to be done, we will need to know what version is the release note being generated for. This can be passed in as a new flag.
